pkg/crc/api/api.go

```diff
@@ -44,18 +48,35 @@ func CreateApiServer(socketPath string) (CrcApiServer, error) {
 }
 
 func (api CrcApiServer) Serve() {
+	go api.handleClusterOperations() // go routine that handles start, stop and delete calls
 	for {
 		conn, err := api.listener.Accept()
 		if err != nil {
 			logging.Error("Error establishing communication: ", err.Error())
 			continue
 		}
-		go api.handleConnection(conn)
+		api.handleConnections(conn) // handle version, status, webconsole, etc. requests
 	}
 }
 
-func (api CrcApiServer) handleConnection(conn net.Conn) {
-	defer conn.Close()
+func (api CrcApiServer) handleClusterOperations() {
+	for req := range api.clusterOpsRequestsChan {
```

Comment:
https://golangr.com/range-channel/ says that buffered channels can only be iterated over after being closed?
You could also go with a `for {}` loop

Reply:
Not sure, this is working as is, maybe this changed now? if the channel is closed the range will throw an error and the go routine dies, anyways we don't want the channel to be closed.

Reply:
Or maybe golangr.com is wrong. From other sites/my testing, closing the channel is needed to stop the loop, but before the channel is closed, this will loop forever.

Reply:
Yes, and will block on the channel read when its empty..